.github/workflows/build-docker.yml

```
@@ -263,20 +268,71 @@ jobs:
           context: .
           project: 7hd4vdzmw5 # astral-sh/uv
           platforms: linux/amd64,linux/arm64
-          push: true
+          push: ${{ needs.docker-plan.outputs.push }}
           tags: ${{ steps.meta.outputs.tags }}
           labels: ${{ steps.meta.outputs.labels }}
+          # TODO(zanieb): Annotations are not supported by Depot yet and are ignored
           annotations: ${{ steps.meta.outputs.annotations }}
 
       - name: Generate artifact attestation
+        if: ${{ needs.docker-plan.outputs.push == 'true' }}
         uses: actions/attest-build-provenance@e8998f949152b193b063cb0ec769d69d929409be # v2.4.0
         with:
           subject-name: ${{ env.UV_GHCR_IMAGE }}
           subject-digest: ${{ steps.build-and-push.outputs.digest }}
 
-  # Re-tag the base image, to ensure it's shown as the newest on the registry UI
-  docker-retag-base:
-    name: retag uv
+      # Push annotations manually.
+      # See `docker-annotate-base` for details.
+      - name: Add annotations to images
+        if: ${{ needs.docker-plan.outputs.push == 'true' }}
+        env:
+          IMAGES: "${{ env.UV_GHCR_IMAGE }} ${{ env.UV_DOCKERHUB_IMAGE }}"
+          DIGEST: ${{ steps.build-and-push.outputs.digest }}
+          TAGS: ${{ steps.meta.outputs.tags }}
+          ANNOTATIONS: ${{ steps.meta.outputs.annotations }}
+        run: |
+          set -x
+          readarray -t lines <<< "$ANNOTATIONS"; annotations=(); for line in "${lines[@]}"; do annotations+=(--annotation "$line"); done
+          for image in $IMAGES; do
+            for tag in $TAGS; do
+              # The tags are fully qualified, so skip the ones that do not match the image
+              if [[ "$tag" != "${image}"* ]]; then
+                continue
+              fi
+              docker buildx imagetools create \
+                "${annotations[@]}" \
+                -t "${tag}" \
+                "${image}@${DIGEST}"
+            done
```

```suggestion
            readarray -t lines < <(grep "^${image}:" <<< "$TAGS"); tags=(); for line in "${lines[@]}"; do tags+=(-t "$line"); done
            docker buildx imagetools create \
              "${annotations[@]}" \
              "${tags[@]}" \
              "${image}@${DIGEST}"
```

Similar to the readarray, but we filter by image. This makes it a single command per image. We could've also used the previous jq command, but we don't have $DOCKER_METADATA_OUTPUT_JSON in the last job so we use readarray to make it the same on both.

e.g.

```bash
          for image in $IMAGES; do
            docker buildx imagetools create \
            $(jq -cr --arg img "${image}" '.tags | map(select(startswith($img)) | "-t " + .) | join(" ")' <<< "$DOCKER_METADATA_OUTPUT_JSON") \
            "${image}@${DIGEST}"
          done
```